File: src/packages/sdk/src/cli/render/snippet.ts
/**
 * Truncate a string for deterministic output.
 * Replaces newlines with space, caps length.
 */

export function truncateSnippet(
	s: string | undefined | null,
	maxLen = 140,
): string {
	if (s == null) return "";
	const normalized = s.replace(/\s+/g, " ").trim();
	if (normalized.length <= maxLen) return normalized;
	return `${normalized.slice(0, maxLen)}…`;
}

File: src/packages/sdk-python/src/evalgate_sdk/cli/render/snippet.py
"""Truncate a string for deterministic output.

Port of ``cli/render/snippet.ts``.
"""

from __future__ import annotations

import re
from typing import Optional


def truncate_snippet(s: Optional[str], max_len: int = 140) -> str:
    """Replaces newlines with space, caps length."""
    if s is None:
        return ""
    normalized = re.sub(r"\s+", " ", s).strip()
    if len(normalized) <= max_len:
        return normalized
    return normalized[:max_len] + "…"
